js_modules/dagster-ui/packages/ui-core/src/hooks/useOpenInNewTab.tsx

```diff
@@ -1,12 +1,16 @@
-import {useContext} from 'react';
+import {useCallback, useContext} from 'react';
 
 import {AppContext} from '../app/AppContext';
 
 // Open a path in a new tab. Incorporates the base path to ensure that
 // the link opens to the appropriate deployment.
 export const useOpenInNewTab = () => {
```

Comment:
Does this need to be a hook? Would prefer a regular function. Maybe we can move the basePath to be returned by `extractCloudDagsterUISettings` if it doesn't already have enough information to obtain it?

Reply:
So currently we pass it to AppContext, maybe we can just use a global instead?

Reply:
I think sometimes you might have logic that is divorced from React (eg: inside some helper function 2 levels deep) and it feels weird to require a component to thread the  `openInNewTab` function that deep. It requires every function along the path to know that this functionality is nested inside which seems like a bad separate of concerns to me.

Reply:
Discussed in Slack, this is a little more awkward in Cloud since we have to manipulate the base path value for dev. Worth coming back to though.